src/components/Dialog/TriggerAndActionDialog/TriggerAndActionDialog.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useCallback, useState } from 'react'
+import styled from 'styled-components'
+
+import { useId } from '../../../hooks/useId'
+import { includeDisabledTrigger } from '../../../libs/util'
+import { ActionDialog } from '../ActionDialog'
+
+type ToggleModalActionType = () => void
+
+export const TriggerAndActionDialog: React.FC<
+  Omit<React.ComponentProps<typeof ActionDialog>, 'isOpen' | 'onClickClose'> & {
+    trigger: React.ReactNode
+    onClickTrigger?: (modalOpen: ToggleModalActionType) => void
+    onClickClose?: (modalClose: ToggleModalActionType) => void
+  }
+> = ({ id, trigger, onClickTrigger, onClickClose, ...props }) => {
+  const generatedId = useId()
+  const actualId = id || generatedId
+  const [isOpen, setIsOpen] = useState(false)
+
+  const modalOpen = useCallback(() => setIsOpen(true), [])
+  const modalClose = useCallback(() => setIsOpen(false), [])
```

Comment:
強い気持ちはありませんが、関数をまとめることもできますかね？
```suggestion
  const toggleModal = useCallback(() => setIsOpen((prev) => !prev), [])
```

Reply:
@yt-ymmt まとめられるメリットはありつつ、結構直感に反する指定になってしまいそうな予感...
(例えばmodalOpenが偶数回連打できてしまうとモーダルが開かないとか)

共通化して得られるメリットは関数一つ分の削除くらいなのでこのままでいいかな、と思っています